api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -65,6 +65,11 @@ message ExtAuthz {
   // request message indicating if the body data is partial.
   BufferSettings with_request_body = 5;
 
+  // Dictates how request body is sent to the external the external authorization service. This is
+  // effective when :ref:`with_request_body <envoy_v3_api_field_extensions.filters.http.ext_authz.v3.ExtAuthz.with_request_body>`
+  // is set to buffer request body.
+  RequestBodyOptions request_body_options = 13;
```

Comment:
Sorry just seeing this now, but since this only matters when using `with_request_body` why not just put the option inside `BufferSettings`?

Reply:
Since https://github.com/envoyproxy/envoy/pull/13131#discussion_r489715036 it was there before. If you think it is better there, I can move it.

Reply:
I would prefer it to be there since it's logically related and they must be set together. @htuch ?

Reply:
Another supporting argument probably, since the field name is `with_request_body` (while the type of it is `BufferSettings`, a bit misleading?). Hence a sample of config will be:

```yaml
  with_request_body:
    max_request_bytes: 10
    allow_partial_message: true
    pack_as_bytes: true
```

Reply:
Yeah let's just do this unless @htuch has a strong objection.

Reply:
Sure, this is fine, objection overruled.